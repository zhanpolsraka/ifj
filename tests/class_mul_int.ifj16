//0
class Main{
	static int res;
	static void run() {
		Main.res = A.val * B.val;
	}
}

class A{
	static int val = 1;
}

class B{
	static int val = 2;
}