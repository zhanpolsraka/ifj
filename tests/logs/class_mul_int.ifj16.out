---------------FILENAME: tests/class_mul_int.ifj16---------------


Instruction stack item [15] -> class with name "Main"
Instruction stack item [14] -> function with name "run"
Instruction stack item [13] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [12] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [11] -> instruction with operation 3 -> MUL
Instruction stack item [10] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [9] -> end function
Instruction stack item [8] -> end class
Instruction stack item [7] -> class with name "A"
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [4] -> end class
Instruction stack item [3] -> class with name "B"
Instruction stack item [2] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [1] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [0] -> end class


---------------MAKE RELATIONS--------------
                     *
                     *
                     *
------------------EXECUTE------------------

------------------OUTPUT-------------------

STEP 1 - non-func instructions
Found Main
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
Found run()
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
Got out of run()
STEP 1 - non-func instructions
Got out of Main
STEP 1 - non-func instructions
STEP 1 - non-func instructions
INSERT 1
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 1 to 0x20c6600
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
INSERT 2
---------------------------------------------------
N0: type is 22, values is 2
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 2 to 0x20c6b90
---------------------------------------------------
N0: type is 22, values is 2
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
STEP 2 - validating existance of run() in Main
STEP 3 - executing run() instructions
INSERT 1
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
INSERT 2
---------------------------------------------------
N1: type is 22, values is 2
N0: type is 22, values is 1
---------------------------------------------------
UNKNOWN INSTRUCTION
---------------------------------------------------
N0: type is 22, values is 2
---------------------------------------------------
ASSIGN 2 to 0x20c6400
---------------------------------------------------
N0: type is 22, values is 2
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------

-----------------END OUTPUT----------------


----------------ALL RIGHT!!!---------------

----------------SYMBOL TREE----------------

Class Main
	Variable -> res, type INTEGER, value [2]
	Function -> run
Class A
	Variable -> val, type INTEGER, value [1]
Class B
	Variable -> val, type INTEGER, value [2]


--------------INSTRUCTION LIST-------------

Instruction stack item [15] -> class with name "Main"
Instruction stack item [14] -> function with name "run"
Instruction stack item [13] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [12] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [11] -> instruction with operation 3 -> MUL
Instruction stack item [10] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [9] -> end function
Instruction stack item [8] -> end class
Instruction stack item [7] -> class with name "A"
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [4] -> end class
Instruction stack item [3] -> class with name "B"
Instruction stack item [2] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [1] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [0] -> end class



